### PR TITLE
Wrapper to use hugging face algorithms for building vocabulary

### PR DESCRIPTION
**Pull request by gusty1g**
# Pull Request Template

## Description

HuggingFace has got algorithms to build vocabulary from a corpus. This pull requests add a wrapper utility to use hugging face vocabulary builders in DeepChem.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by gusty1g** [comment]
Note: I am yet to add docstring and docs.

File: deepchem/feat/vocabulary_builders/hf_vocab.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+from deepchem.feat.vocabulary_builders.vocabulary_builder import VocabularyBuilder
+from tokenizers import Tokenizer
```
Comment (rbharath) [inline deepchem/feat/vocabulary_builders/hf_vocab.py:2]: Is huggingface already an optional dependency?

File: deepchem/feat/vocabulary_builders/hf_vocab.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+from typing import List
+
+
+class HuggingFaceVocabularyBuilder(VocabularyBuilder):
```
Comment (rbharath) [inline deepchem/feat/vocabulary_builders/hf_vocab.py:6]: Can you add this to the docs?

**Review by rbharath — COMMENTED** [review]
Took a preliminary look and everything looks ok at first glance, but will hold off on a full review until docstrings/examples are all added in

File: deepchem/feat/vocabulary_builders/hf_vocab.py (hunk 1 above)
Comment (gusty1g) [reply deepchem/feat/vocabulary_builders/hf_vocab.py:2]: yes

File: deepchem/feat/vocabulary_builders/hf_vocab.py (hunk 2 above)
Comment (gusty1g) [reply deepchem/feat/vocabulary_builders/hf_vocab.py:9]: added to docs 42479ce

File: requirements/env_common.yml [hunk 3]
```diff
@@ -22,7 +22,8 @@ dependencies:
     - pyGPGO
     - pymatgen
     - simdna
-    - transformers==4.10.*
+    - transformers
+    - tokenizers
```
Comment (rbharath) [inline requirements/env_common.yml:26]: If `tokenizers` is a new soft requirement, we should add it to the soft requirements table in the docs.

**Review by rbharath — COMMENTED** [review]
Almost there; one minor request for the docs and we should be good to go

**Review by rbharath — APPROVED** [review]
LGTM. Please feel free to merge once CI looks good

**Comment by gusty1g** [comment]
The CI is failing in installation of `transformers` for python 3.10 but it is installing in python 3.8.

**Pull request merged**